gcloud/bigtable/cluster.py

```diff
@@ -15,9 +15,51 @@
 """User friendly container for Google Cloud Bigtable Cluster."""
 
 
+import re
+
 from gcloud.bigtable.table import Table
 
 
+_CLUSTER_NAME_RE = re.compile(r'^projects/(?P<project>[^/]+)/'
+                              r'zones/(?P<zone>[^/]+)/clusters/'
+                              r'(?P<cluster_id>[a-z][-a-z0-9]*)$')
+
+
+def _require_pb_property(message_pb, property_name, value):
```

Comment:
The other option would be to split the funciton in two (since there are two purposes):

```python
def _get_pb_property_value(message_pb, property_name)
    """Return a message field value."""
    all_fields = set([field.name for field in message_pb._fields])
    if property_name not in all_fields:
        raise ValueError('Message does not contain %s.' % (property_name,))
    return getattr(message_pb, property_name)

def _assert_pb_property_value(message_pb, property_name, expected_value):
    """Raise if pb field does not exist, or does not match `expected_value`."""
    if _get_pb_property_value(message_pb, property_name) != expected_value:
        raise ValueError('Message returned %s value disagreeing '
                         'with value passed in.' % (property_name,))
```